### PR TITLE
POC patch separation

### PR DESCRIPTION
**Pull request by estellecomment**
An attempt to separate out our code into different patches, and keep things understandable and maintainable.

Create a patch of matrix-react-sdk for a new feature  :
`patch-package --patch-dir patches/myfeature --include path/to/react-sdk/files/modified  matrix-react-sdk`
(the `--include` arg is so that only your modification is included in your patch. By default, patch-package throws all modifications into the same file, which is harder to keep track.)

This creates the file : `patches/myfeature/matrix-react-sdk+3.53.0.patch` (or whatever version number you are patching)

Then, the postinstall script will apply each patch separately, you don't need to do anything. 
It will run: `patch-package --patch-dir patches/myfeature`

Edit : if you are a human, you can view patch files in Vscode with this extension for example : 
Diff Viewer
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=caponetto.vscode-diff-viewer

### CODE REVIEW

File: patches/joinrules/matrix-react-sdk+3.53.0.patch [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+-                { upgradeRequiredPill }
+-            </>,
+-            description,
+-            // if there are 0 allowed spaces then render it as invite only instead
+-            checked: joinRule === JoinRule.Restricted && !!restrictedAllowRoomIds?.length,
+-        });
+-    }
++    /**
++     * :TCHAP: removed all code about "restricted" sharing, which has to do with spaces.
++     * See patches/matrix-react-sdk+<version>.patch for details.
```
Comment (estellecomment) [inline patches/joinrules/matrix-react-sdk+3.53.0.patch:112]: leave notes in the patched files so we know what it is about.

Comment (odelcroi) [reply patches/joinrules/matrix-react-sdk+3.53.0.patch:112]: Is there additional notes in the patch? I don't see them

Comment (estellecomment) [reply patches/joinrules/matrix-react-sdk+3.53.0.patch:112]: It's just that "TCHAP" comment. The patch is generated, so you can't really add comments directly in it. So I added the comments in the code file.

**Comment by odelcroi** [comment]
Thanks for the poc, I have a PR using it here : https://github.com/tchapgouv/tchap-web-v4/pull/174
I guess this PR can be closed

**Pull request closed without merging**